src/logging_unittest.cc

```diff
@@ -995,6 +997,96 @@ static void TestCustomLoggerDeletionOnShutdown() {
   EXPECT_FALSE(IsGoogleLoggingInitialized());
 }
 
+namespace LogTimes {
+// Log a "message" every 10ms, 10 times. These numbers are nice compromise
+// between total running time of 100ms and the period of 10ms. The period is
+// large enough such that any CPU and OS scheduling variation shouldn't affect
+// the results from the ideal case by more than 5% (500us or 0.5ms)
+GLOG_CONSTEXPR double LOG_PERIOD_SEC = 0.01;
+GLOG_CONSTEXPR int64_t LOG_PERIOD_US     = 10000;
+GLOG_CONSTEXPR int64_t LOG_PERIOD_TOL_US = 500;
+
+// Set an upper limit for the number of times the stream operator can be
+// called. Make sure not to exceed this number of times the stream operator is
+// called, since it is also the array size and will be indexed by the stream
+// operator.
+GLOG_CONSTEXPR size_t MAX_CALLS          = 10;
+}  // namespace LogStreamTimes
+
+#if HAVE_CXX11_CHRONO
+struct LogTimeRecorder {
+  size_t m_streamTimes = 0;
+  std::chrono::steady_clock::time_point m_callTimes[LogTimes::MAX_CALLS];
+};
+// The stream operator is called by LOG_EVERY_T every time a logging event
+// occurs. Make sure to save the times for each call as they will be used later
+// to verify the time delta between each call.
+std::ostream& operator<<(std::ostream& stream, LogTimeRecorder& t) {
+  t.m_callTimes[t.m_streamTimes++] = std::chrono::steady_clock::now();
+  return stream;
+}
+// get elapsed time in microseconds
+int64 elapsedTime_us(const std::chrono::steady_clock::time_point& begin,
+        const std::chrono::steady_clock::time_point& end) {
+  return std::chrono::duration_cast<std::chrono::microseconds>((end - begin))
+          .count();
+}
+#elif defined(OS_WINDOWS)
+struct LogTimeRecorder {
+  size_t m_streamTimes = 0;
+  LARGE_INTEGER m_callTimes[LogTimes::MAX_CALLS];
+};
+std::ostream& operator<<(std::ostream& stream, LogTimeRecorder& t) {
+  QueryPerformanceCounter(&t.m_callTimes[t.m_streamTimes++]);
+  return stream;
+}
+// get elapsed time in micrseconds
+int64 elapsedTime_us(const LARGE_INTEGER& begin, const LARGE_INTEGER& end) {
+  LARGE_INTEGER freq;
+  QueryPerformanceFrequency(&freq);
+  return (end.QuadPart - begin.QuadPart) * 1000000 / freq.QuadPart;
+}
+#else
+struct LogTimeRecorder {
+  size_t m_streamTimes = 0;
+  timespec m_callTimes[LogTimes::MAX_CALLS];
+};
+std::ostream& operator<<(std::ostream& stream, LogTimeRecorder& t) {
+  clock_gettime(CLOCK_MONOTONIC, &t.m_callTimes[t.m_streamTimes++]);
+  return stream;
+}
+// get elapsed time in microseconds
+int64 elapsedTime_us(const timespec& begin, const timespec& end) {
```

Comment:
This is something I previously overlooked. Wouldn't it make sense to always you the highest possible precision in ns?

Reply:
Not for this particular test, because the error tolerance is 500us and any variation in the hundreds of nanoseconds is insignificant. That being said, I'll go back and change to nanoseconds to remove some of the math just to simply the functions.